pkg/rbac/middleware.go

```diff
@@ -0,0 +1,77 @@
+package rbac
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+//verifies token (userid) with the TokenReview:
+func Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
```

Comment:
Confused by multiple returns here . Couldnt we have `func Middleware(next http.Handler) http.Handler {` @Anxhela21 @jlpadilla

Reply:
I don't think so because in server.go we call the middleware using `router.Use(...)`
(defined as: `func (*mux.Router).Use(mwf ...mux.MiddlewareFunc)`) which expects the middleware to be of type `mux.MiddlewareFunc` so if we call using: router.Use(rbac.Middleware(next)) to return http.Handler type we get an error.

Reply:
Not sure if this will work, but you can try this.
```
router.Use(rbac.Middleware)
```

Then here
```
func Middleware() func(next http.Handler) http.Handler {
```

Reply:
that should work as long as we return _a function returning the http.Handler_ instead of directly returning one with Middleware()